asciidoctor-tests/TestAdocLinks.lua

```diff
@@ -0,0 +1,125 @@
+-- TestAdocLinks.lua - a test to verify that all external links in adoc files are functional
+-- Copyright (C) 2014-2016 Zach Rhoads
+
+-- This program is free software:  you can redistribute it and/or modify it
+-- under the terms of  the  GNU General Public License  as published by the
+-- Free Software Foundation, version 3 of the License.
+--
+-- This program  is  distributed  in the hope  that it will be useful,  but
+-- WITHOUT  ANY WARRANTY;  without  even the implied warranty of MERCHANTA-
+-- BILITY or  FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public
+-- License for more details.
+--
+-- You should have received a copy of the GNU General Public License  along
+-- with this program. If not, see <http://www.gnu.org/licenses/>.
+
+TestAdocLinks = {
+	metadata = {
+		description = "Verify that all external links in adoc files are functional",
+		authors = "Zach Rhoads",
+		changed = "2016-07-15",
+		tags = {"AsciiDoc"}
+	},
+	requires = {curl},
+	asciiObj = nil,
+	main_file = nil,
+  files = {},
+  links = {}
+}
+
+
+-- flattens tree to a map by doc name
+function TestAdocLinks.walkContentTree(current)
+
+  TestAdocLinks.files[current.file] = current.content
+
+  if current.children then
+    for key,value in pairs(current.children) do
+      TestAdocLinks.walkContentTree(value)
+    end
+  end
+end
+
+
+-- finds links in each line of content in each file
+-- matches both adoc-style links and md-style links
+-- NOTE: all adoc-style links will match twice since they overlap with md-style links, but each link will only be tested once
+function TestAdocLinks.findLinks()
+
+  pattern = "link:https?://[%a%d%./_=%?%-#]*%["
+  mdPattern =    "https?://[%a%d%./_=%?%-#]*%["
+  
+  skipLinks = {
+    ["http://skip-this-site.com"] = true,
```

Comment:
This seems to be a good place to include some other frequently used [reserved domains](https://tools.ietf.org/html/rfc2606):

- example.com
- example.net
- example.org
- example.edu

I would also recommend skipping local hostnames:

- localhost
- 127.0.0.1

Personally, I would also consider allowing the use of patterns in skipped links to whitelist subdomains like `something.example.com`.